packages/mui-system/src/Stack/createStack.tsx

```diff
@@ -141,6 +141,9 @@ export const style = ({ ownerState, theme }: StyleFunctionProps) => {
             breakpoint ? directionValues[breakpoint] : ownerState.direction,
           )}`]: getValue(transformer, propValue),
         },
+        '& > :not([style]):first-child': {
```

Comment:
added this new selector which adds `margin:0` to the first child if it is not a `style` attribute.

Reply:
Could you explain the reasoning behind `:not([style])`?

Reply:
so the exsisting selector targeted all elements except for `style` element [here](https://github.com/mui/material-ui/blob/2f0c21e30fee821e31beec376695eb93e75bdb60/packages/mui-system/src/Stack/createStack.tsx#L138C1-L138C1) , so i followed same convention here.

Reply:
okay just realized, it should be :not(style) , pushing the fix

Reply:
I'm afraid this won't work well either - if the first element is indeed `style`, the next element (which should have the margin reset) won't be targeted by the `:first-child` selector, as it's not the first one.

Reply:
yup you are right. updated selector to `& > :not(style):first-of-type` which selects first element which is not `style`. i belive this selector should do the trick.

here is the playground if you want to test out other scenarios: https://codepen.io/sai6855/pen/GRwRGwJ

Reply:
This also could be problematic if there are different tags inside the stack (for example, `p` and `div`) - this selector will target the first tag of each type.

Wouldn't removing `margin: 0` from the original code work well enough? We could reset the top and bottom margins and set the top margin to `8px` as it was before.

Reply:
I suspect that

```diff
diff --git a/packages/mui-system/src/Stack/createStack.tsx b/packages/mui-system/src/Stack/createStack.tsx
index 7109553614..14b2b3ee92 100644
--- a/packages/mui-system/src/Stack/createStack.tsx
+++ b/packages/mui-system/src/Stack/createStack.tsx
@@ -135,8 +135,10 @@ export const style = ({ ownerState, theme }: StyleFunctionProps) => {
         return { gap: getValue(transformer, propValue) };
       }
       return {
-        '& > :not(style) + :not(style)': {
+        '& > :not(style):not(style)': {
           margin: 0,
+        },
+        '& > :not(style) ~ :not(style)': {
           [`margin${getSideFromDirection(
             breakpoint ? directionValues[breakpoint] : ownerState.direction,
           )}`]: getValue(transformer, propValue),
```

could solve both #37381 and #37338.

Reply:
What's the purpose duplication in `:not(style):not(style)`?

Reply:
to win over `& > :not(style) ~ :not(style)` of previous responsive values